app/main.py

```diff
@@ -1,3 +1,38 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        file_config = json.load(file)
+
+    for customer in file_config["customers"]:
+        buyer = Customer(
+            customer["name"],
+            customer["product_cart"],
+            customer["location"],
+            customer["money"],
+            customer["car"]["fuel_consumption"]
+        )
+
+        buyer.print_customer_name_and_money()
+        price_in_shops = {}
+        for shop in file_config["shops"]:
+            shop = Shop(shop["name"], shop["location"], shop["products"])
+            price = buyer.get_price_product_cart(shop, file_config)
+            price_in_shops[shop] = price
```

Comment:
Using `Shop` objects as keys in the `price_in_shops` dictionary may lead to issues because objects are not hashable by default. Consider using the shop's name as the key instead.